File: Android/karaokekit/src/main/java/com/netease/yunxin/kit/karaokekit/api/NEKaraokeCopyrightedMediaListener.kt
/*
 * Copyright (c) 2022 NetEase, Inc. All rights reserved.
 * Use of this source code is governed by a MIT license that can be
 * found in the LICENSE file.
 */

package com.netease.yunxin.kit.karaokekit.api

/** 下载回调  */
interface NEKaraokeCopyrightedMediaListener {
    /**
     * 开始下载
     *
     * @param songId 歌曲编号
     */
    fun onPreloadStart(songId: String?, channel: Int)

    /**
     * 下载进度更新
     *
     * @param songId 歌曲编号
     * @param progress 进度
     */
    fun onPreloadProgress(songId: String?, channel: Int, progress: Float)

    /**
     * 下载完成
     *
     * @param songId 歌曲编号
     * @param errorCode 错误码
     * @param msg 错误信息
     */
    fun onPreloadComplete(songId: String?, channel: Int, errorCode: Int, msg: String?)
}

File: iOS/KaraokeKit/NEKaraokeKit/NEKaraokeKit/Classes/Public/NEKaraokeCopyrightedMediaListener.swift
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

import Foundation
import NECopyrightedMedia
@objc
public protocol NEKaraokeCopyrightedMediaListener: NSObjectProtocol {
  /// 开始下载回调
  /// @param songId 歌曲Id
  @objc optional func karaoke_onPreloadStart(_ songId: String, channel: SongChannel)

  /// 下载进度回调
  /// @param songId 歌曲ID
  /// @param progress 进度值
  @objc optional func karaoke_onPreloadProgress(_ songId: String, channel: SongChannel,
                                                progress: Float)

  /// 下载完成
  /// @param songId 歌曲歌曲ID
  /// @param error 成功为nil
  @objc optional func karaoke_onPreloadComplete(_ songId: String, channel: SongChannel,
                                                error: Error?)
}

@objc
public protocol NEKaraokeCopyrightedMediaEventHandler: NSObjectProtocol {
  /// Tokne过期
  @objc optional func onTokenExpired()
}
